.eslintrc.js

```diff
@@ -28,182 +12,202 @@ module.exports = {
     "globalThis": false,
   },
   reportUnusedDisableDirectives: true,
-  rules: {
-    "@typescript-eslint/adjacent-overload-signatures": "warn",
-    "@typescript-eslint/array-type": [
-      "warn",
-      {
-        default: "array-simple",
-        readonly: "generic"
-      },
-    ],
-    "@typescript-eslint/ban-ts-comment": "warn",
-    "@typescript-eslint/ban-types": [
-      "warn",
-      {
-        "types": {
-          "{}": false,
-        },
-        extendDefaults: true,
-      }
+  overrides: [{
+    files: ["*.ts", "*.tsx"],
+    parserOptions: {
```

Comment:
Its hard to tell whats going on in this file since so much was moved around but previously we were setting `parserOptions` for the whole linting project and now its only in this `overrides` section?

Reply:
Basically since we have a couple regular .js files and our entire linting architecture was set up to only lint typescript, I moved the ts specific stuff to `overrides` so that we could lint the whole project including files that weren't in the tsconfig files (including js files in the tsconfig can lead to some messy issues)